File: Data Structures & Algorithms/remove-element/submission-3.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int valid = 0;
        for(int i=0; i<nums.size();i++){
            if(nums[i] != val){
                nums[valid] = nums[i];
                valid++;
            }
        }
        return valid;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-3.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        pre = []
        mult = 1;
        for i in nums:
            pre.append(mult)
            mult *= i

        post = []
        mult = 1;
        for i in range(len(nums)-1,-1,-1):
            post.append(mult)
            mult *= nums[i]

        post.reverse()
        ans = [0] * len(pre)
        for i in range(len(pre)):
            ans[i] = pre[i] * post[i]

        return ans